### PR TITLE
DEC-446 Add Browserify to project

### PR DESCRIPTION
**Pull request by danthewolfe**
Adding in Browserify to make it easier to make it easier to include external libraries.

Followed adaptations from https://reactjsnews.com/setting-up-rails-for-react-and-jest/ and Beehive's setup.

### CODE REVIEW

File: .hound.yml [hunk 1]
```diff
@@ -3,4 +3,5 @@ ruby:
 
 javascript:
   config_file: config/style_guides/.javascript-style.json
+  ignore_file: app/assets/javascripts/components.js
```
Comment (jaronkk) [inline .hound.yml]: I think this is expecting to find a file that specifies a list of javascript files to ignore.  Beehive example: https://github.com/ndlib/beehive/blob/master/config/style_guides/.javascript-ignore

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,16 @@
 language: ruby
-cache: bundler
 sudo: false
 rvm:
   - 2.1.4
+install:
+  - bundle install
```
Comment (jaronkk) [inline .travis.yml]: We don't want to specify bundle install, since I think it overrides the way Travis does bundle caching between builds.

**Comment by jaronkk** [comment]
:+1:

**Comment by jaronkk** [comment]
The deployment depends on node being installed on the servers, but I think we can go ahead and merge this. :+1:

**Comment by jonhartzler** [comment]
:+1:

**Pull request merged**